### PR TITLE
Reaction log

### PR DESCRIPTION
**Pull request by Ailim**
Keeps log of reaction usage in local database. Adds command !reactions :reaction: which shows how many times the reaction has been used, and !topreactions, which lists the most used reactions.

Reactions in DMs or on own messages are not counted.

### CODE REVIEW

File: src/plugins/eventlog.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	}
+}
+
+function isPrivateOrOwnMessage(reaction, user) {
+	return reaction.message.author.id === user.id || reaction.message.channel.type === 'dm';
+}
+
+export default async function eventlog(discord) {
+	log.info('starting eventlog plugin');
+	discord.on('raw', async (event) => {
```
Comment (AlexKolpa) [inline src/plugins/eventlog.js]: Could you clarify what this raw event is for?

File: src/plugins/eventlog.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+			const message = await channel.fetchMessage(data.message_id);
+			const emojiKey = (data.emoji.id) ? `${data.emoji.name}:${data.emoji.id}` : data.emoji.name;
+			let reaction = message.reactions.get(emojiKey);
+			if (!reaction) {
+				// Create an object that can be passed through the event like normal
+				const emoji = new Discord.Emoji(discord.guilds.get(data.guild_id), data.emoji);
+				reaction = new Discord.MessageReaction(message, emoji, 1, data.user_id === discord.user.id);
+			}
+			discord.emit(events[event.t], reaction, user);
```
Comment (AlexKolpa) [inline src/plugins/eventlog.js]: What's this emit for?

File: src/plugins/eventlog.js (hunk 1 above)
Comment (Ailim) [reply src/plugins/eventlog.js]: The `messageReactionAdd` and `messageReactionRemove` we are listening for trigger only for messages that have been cached. Raw listens to all events that happen. From these raw events, we can parse reaction add/delete events also on messages that haven't been cached, and emit a new event that can then be handled by the two listeners.

File: src/plugins/eventlog.js (hunk 2 above)
Comment (Ailim) [reply src/plugins/eventlog.js]: Basically are converting reaction add/delete events from the raw feed to events that can be handled by our two reaction event listeners.

**Comment by Ailim** [comment]
> 
> 
> Looks good, but I feel like the raw block could do with some documentation

Added some documentation.

File: src/plugins/eventlog.js (hunk 1 above)
Comment (AlexKolpa) [reply src/plugins/eventlog.js]: ah cool, yeah that makes sense :+1:

**Pull request merged**